modules/core/src/main/java/org/apache/fluo/core/impl/TransactionImpl.java

```diff
@@ -641,16 +634,8 @@ private void checkForOrphanedReadLocks(CommitData cd, Map<Bytes, Set<Column>> lo
   }
 
   private void checkForOrphanedLocks(CommitData cd) throws Exception {
-
-    Map<Bytes, Set<Column>> locksSeen = new HashMap<>();
-
-    readUnread(cd, kve -> {
```

Comment:
I think passing the locksSeen consumer is still needed.  The intention behind this code was to record all write locks resolved so those could be avoided when processing read locks.  So I think ParallelSnapshot scanner should take a writeLocksSeen parameter in addition to readLocksSeen.  It can populate   writeLocksSeen like SnapshotScanner was doing.   

In `ParallelSnapshotScanner.scan(Map, List)` I think the following change could be made.

```java
// snippet from  scan(...) method
          case LOCK:
            locks.add(entry);
            // add following line.. and also add writeLocksSeen instance var
            writeLocksSeen.accept(entry);
            break;
// end snippet
```

Sorry I didn't notice the difference between the read and write locks seen when you asked about this on the issue.  Only noticed it when taking a deeper dive.

Reply:
I don't think I fully understand, is the idea to avoid all write locks or is it that only a portion of the write locks are actually resolved and we should avoid that subset?

In terms of keeping track of write locks in `ParallelSnapshotScanner` would it be more consistent to add another `private Map<Bytes, Set<Column>>` similar to how read locks are tracked? Instead of adding a Consumer.

Also I didn't know that I could change the public constructors of `SnapshotScanner` and `ParallelSnapshotScanner`, or should I just override them? Is their any documentation I'd have to update if I change the constructors?

Reply:
> I don't think I fully understand, is the idea to avoid all write locks or is it that only a portion of the write locks are actually resolved and we should avoid that subset?

The goal is to avoid work.  If a write lock was resolved for row/column, then we know there was not a read lock at that row column.  Therefore no work needs to be done attempting to resolve a read lock.

> In terms of keeping track of write locks in ParallelSnapshotScanner would it be more consistent to add another private Map<Bytes, Set<Column>> similar to how read locks are tracked? Instead of adding a Consumer.

I think the consumer was used to avoid work.  Tracking of write locks is not always needed. So when its not needed there is no need to update a map.  That is why the code would sometimes pass in a no-op lambda for the consumer.

> Also I didn't know that I could change the public constructors of SnapshotScanner and ParallelSnapshotScanner, or should I just override them? Is their any documentation I'd have to update if I change the constructors?

Those classes are not public API, so you can make whatever changes you like.

Reply:
Those for the explanation, I've made the changes.